lisp/ess-r-d.el

```diff
@@ -190,16 +190,57 @@
 (easy-menu-add-item inferior-ess-mode-menu nil ess-tracebug-menu "end-dev")
 
 
-;; modify S Syntax table:
-(setq R-syntax-table S-syntax-table)
+;; Inherit from the S syntax table:
+(setq ess-r-syntax-table (copy-syntax-table S-syntax-table))
+
+;; Letting Emacs treat backquoted names and %ops% as strings solves
+;; many problems with regard to nested strings and quotes
+(modify-syntax-entry ?` "\"" ess-r-syntax-table)
+(modify-syntax-entry ?% "\"" ess-r-syntax-table)
+
+;; Underscore is valid in R symbols
+(modify-syntax-entry ?_ "_" ess-r-syntax-table)
+
+(modify-syntax-entry ?: "." ess-r-syntax-table)
+(modify-syntax-entry ?@ "." ess-r-syntax-table)
+(modify-syntax-entry ?$ "." ess-r-syntax-table)
+
+;; Prevent string delimiting characters from messing up output in the
+;; inferior buffer
+(setq inferior-ess-r-syntax-table (copy-syntax-table ess-r-syntax-table))
+(modify-syntax-entry ?\' "." inferior-ess-r-syntax-table)
+(modify-syntax-entry ?\" "." inferior-ess-r-syntax-table)
+(modify-syntax-entry ?` "." inferior-ess-r-syntax-table)
+(modify-syntax-entry ?% "." inferior-ess-r-syntax-table)
+
+(defun ess-r-font-lock-syntactic-face-function (state)
```

Comment:
@lionel-, I am extracting indentation and font-lock pieces into a stand alone r-mode, and I wonder if this piece is strictly necessary and how it actually works. 

First of all, can we simply include backquote into our function regexp and remove the `font-lock-function-name-face` from here? Secondly, is does this actually work as expected? Both backquote and `%` are punctuation characters in ess-r-mode and if  I understand correctly this function will never be called with `%op%` and backquoted names.